tests/rados/test_rados_preempt_scrub.py

```diff
@@ -0,0 +1,198 @@
+    3.Start writing the data into pool and start scrub parallely
+    4.Once the scrub started check the OSD logs contain the-
+       - head preempted
+       - WaitReplicas::react(const GotReplicas&) PREEMPTED
+    5. Once the tests are complete delete the pool and remove the set values of the parameter
+    """
+
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_object = RadosOrchestrator(node=cephadm)
+    mon_obj = MonConfigMethods(rados_obj=rados_object)
+    ceph_nodes = kw.get("ceph_nodes")
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    pool_target_configs = config["verify_osd_preempt"]["configurations"]
+
+    try:
+        scrub_status_flag = False
+        # Before starting the tests check that scrubbing is not progress in the cluster
+        time_end = time.time() + 60 * 15
+        while time.time() < time_end:
+            scrub_status = check_scrub_status(rados_object)
+            if scrub_status is False:
+                scrub_status_flag = True
+                log.info("Scrubbing is not in progress.Continuing the test execution")
+                break
+            log.info("Waiting for the scrub to complete")
+            time.sleep(30)
+
+        if scrub_status_flag is False:
+            log.error(
+                "Scrubbing is in progress cannot continue the further execution of test"
+            )
+            return 1
+
+        for entry in pool_target_configs.values():
+            log.debug(
+                f"Creating {entry['pool_type']} pool on the cluster with name {entry['pool_name']}"
+            )
+            method_should_succeed(
+                rados_object.create_pool,
+                **entry,
+            )
+
+            log.info(
+                f"Created the pool {entry['pool_name']}. beginning to create large number of omap entries on the pool"
+            )
+            rados_object.bench_write(
+                pool_name=entry["pool_name"], rados_write_duration=30
+            )
+
+            mon_obj.set_config(
+                section="osd", name="osd_shallow_scrub_chunk_max", value="250"
+            )
+            mon_obj.set_config(section="osd", name="osd_scrub_chunk_max", value="250")
+            mon_obj.set_config(section="osd", name="debug_osd", value="10")
+
+            rados_object.bench_write(
+                pool_name=entry["pool_name"], rados_write_duration=60, background=True
+            )
+
+            log_lines = [
+                "head preempted",
+                "WaitReplicas::react(const GotReplicas&) PREEMPTED",
+            ]
+
+            init_time, _ = installer.exec_command(cmd="sudo date '+%Y-%m-%d %H:%M:%S'")
+            rados_object.run_scrub(pool=entry["pool_name"])
+            scrub_status_flag = False
+
+            # Wait for the scrub to start for 15 minutes
+            time_end = time.time() + 60 * 15
```

Comment:
is 15 minute enough for cluster to initiate and complete scrubbing?

Reply:
Yes, I executed a couple of runs and it is working fine.